src/screens/HyperSwitch/PaymentLogs/PaymentLogs.res

```diff
@@ -323,139 +362,165 @@ let make = (~paymentId, ~createdAt) => {
         <div className="flex flex-col w-1/2 gap-12 overflow-y-scroll">
           <p className="text-lightgray_background font-semibold text-fs-16">
             {"Audit Trail"->React.string}
           </p>
           <div className="flex flex-col">
-            {allLogsData
+            {logs.contents
             ->Array.mapWithIndex((paymentDetailsValue, index) => {
               <ApiDetailsComponent
                 key={index->string_of_int}
                 paymentDetailsValue={paymentDetailsValue->getDictFromJsonObject}
-                setResponseObject
-                setRequestObject
-                currentSelected
-                setCurrentSelected
-                setCurrentSelectedType
+                setLogDetails
+                setSelectedOption
+                currentSelected=selectedOption.value
                 paymentId
                 index
-                logsDataLength={allLogsData->Array.length - 1}
+                logsDataLength={logs.contents->Array.length - 1}
               />
             })
             ->React.array}
           </div>
         </div>
         <UIUtils.RenderIf
-          condition={responseObject->String.length > 0 || requestObject->String.length > 0}>
+          condition={logDetails.response->String.length > 0 ||
+            logDetails.request->String.length > 0}>
           <div
             className="flex flex-col gap-4 bg-hyperswitch_background rounded show-scrollbar scroll-smooth overflow-scroll px-8 py-4 w-1/2">
-            <UIUtils.RenderIf condition={requestObject->String.length > 0}>
+            <UIUtils.RenderIf
+              condition={!(logDetails.request->isEmptyString) &&
+              selectedOption.optionType !== WEBHOOKS}>
               <PrettyPrintJson
-                jsonToDisplay=requestObject
-                headerText={Some(currentSelectedType === Payment ? "Request body" : "Event")}
-                maxHeightClass={responseObject->String.length > 0 ? "max-h-25-rem" : ""}
+                jsonToDisplay=logDetails.request
+                headerText={Some(selectedOption.optionType === PAYMENTS ? "Request body" : "Event")}
+                maxHeightClass={logDetails.response->String.length > 0 ? "max-h-25-rem" : ""}
               />
             </UIUtils.RenderIf>
-            <UIUtils.RenderIf condition={responseObject->String.length > 0}>
-              <PrettyPrintJson
-                jsonToDisplay=responseObject
-                headerText={Some(currentSelectedType === Payment ? "Response body" : "Metadata")}
-              />
+            <UIUtils.RenderIf condition={!(logDetails.response->isEmptyString)}>
+              {
+                let headerText = switch selectedOption.optionType {
+                | PAYMENTS => "Response body"
+                | WEBHOOKS => "Request body"
+                | SDK => "Metadata"
+                }->Some
```

Comment:
Can this be put outside JSX?

`  let headerText = switch selectedOption.optionType {
                | PAYMENTS => "Response body"
                | WEBHOOKS => "Request body"
                | SDK => "Metadata"
                }->Some`
                
And then {} will be removed